src/vs/workbench/contrib/testing/browser/codeCoverageDecorations.ts

```diff
@@ -249,6 +249,105 @@ export class CodeCoverageDecorations extends Disposable implements IEditorContri
+	 * @returns true if navigation occurred, false if no missed line was found
+	 */
+	public goToNextMissedLine(): boolean {
+		return this.navigateToMissedLine(true);
+	}
+
+	/**
+	 * Navigate to the previous missed (uncovered) line from the current cursor position.
+	 * @returns true if navigation occurred, false if no missed line was found
+	 */
+	public goToPreviousMissedLine(): boolean {
+		return this.navigateToMissedLine(false);
+	}
+
+	private isMissedLine(detail: CoverageDetailsWithBranch): boolean {
+		if (detail.type === DetailType.Statement) {
+			return !detail.count;
+		} else if (detail.type === DetailType.Branch) {
+			return !detail.detail.branches![detail.branch].count;
+		}
+		return false;
+	}
+
+	private navigateToMissedLine(next: boolean): boolean {
+		const model = this.editor.getModel();
+		const position = this.editor.getPosition();
+		if (!model || !position || !this.details) {
+			return false;
+		}
+
+		// Collect all missed line decorations
+		const missedLines: { lineNumber: number; range: Range }[] = [];
+		for (const [, { detail, options }] of this.decorationIds) {
+			const isMissed = detail.metadata.detail.type === DetailType.Statement
+				? !detail.metadata.detail.count
+					: detail.metadata.detail.type === DetailType.Branch
+						? (() => {
+							const branchDetail = detail.metadata.detail.detail;
+							const branchIndex = detail.metadata.detail.branch;
+							const branches = branchDetail.branches;
+							if (branches && branches[branchIndex]) {
+								return !branches[branchIndex].count;
+							}
+							return false;
+						})()
+						: false;
+			// Check if this is a missed line (CLASS_MISS in lineNumberClassName)
+			if (options.lineNumberClassName?.includes(CLASS_MISS)) {
+				const range = detail.range;
+				const isMissed = this.isMissedLine(detail.metadata.detail);
+
+				if (isMissed && !range.isEmpty()) {
+					missedLines.push({ lineNumber: range.startLineNumber, range });
+				}
+			}
+		}
+
+		// Sort missed lines by line number
+		missedLines.sort((a, b) => a.lineNumber - b.lineNumber);
+
+		if (missedLines.length === 0) {
+			return false;
+		}
+
+		const currentLine = position.lineNumber;
+		let targetLine: { lineNumber: number; range: Range } | undefined;
+
+		if (next) {
+			// Find the first missed line after the current line
+			targetLine = missedLines.find(l => l.lineNumber > currentLine);
+			// If no line found after, wrap around to the first missed line
+			if (!targetLine) {
+				targetLine = missedLines[0];
+			}
+		} else {
+			// Find the last missed line before the current line
+			for (let i = missedLines.length - 1; i >= 0; i--) {
+				if (missedLines[i].lineNumber < currentLine) {
+					targetLine = missedLines[i];
+					break;
+				}
+			}
+			// If no line found before, wrap around to the last missed line
+			if (!targetLine) {
+				targetLine = missedLines[missedLines.length - 1];
+			}
+		}
```

Comment:
You could avoid by just keeping track of the closest missed coverage decoration seen before/after the current position in the earlier loop.